### ISSUE TITLE
/usr/bin/ar: permission denied when running haste-boot

### ISSUE THREAD

**Issue by simonjrp** [valderman/haste-compiler#353]
I can't run haste-boot after a fresh installation on El Capitan (via cabal). I immediately get "/usr/bin/ar: permission denied", even when running as root. This might be related to the ["System Integrity Protection"] (https://support.apple.com/en-us/HT204899) that Apple introduced in OS X El Capitan.

Edit: Seems to work if one disables SIP before running haste-boot (boot into safe mode and run "csrutil disable"), and then re-enables it again afterwords.

**Comment by valderman** [comment]
That's interesting, and slightly weird: Haste should never be calling `ar`. Does your `.cabal/config` contain `shared: True` or any other interesting flags? This seems like it may be related to, or at least the same kind of problem as, #348.

**Comment by bananu7** [comment]
It's called by `haste-boot` which calls `haste-cabal` which tries to call `ar`. I can, sadly, reproduce. No `shared-true`.

And I'm pretty sure it's related to [this main cabal issue](https://github.com/haskell/cabal/issues/2653)

**Comment by bananu7** [comment]
Building haste-cabal fails with:

    Resolving dependencies...

    Distribution/Simple.hs:106:8:
        Could not find module ‘System.Directory’
        Use -v to see a list of the files searched for.

    Distribution/Simple/Utils.hs:199:8:
        Could not find module ‘System.Process’
        Use -v to see a list of the files searched for.

During the config step. Seems like it's a no-go at this moment :(.

**Comment by atamon** [comment]
@bananu7 I've run into the same issue it seems. I tried building cabal from source as stated in the README, and it includes unix 2.7.1.0 when run. But I still get the same error as above.

Did you get this working, if so, how? :)

Cheers

**Comment by bananu7** [comment]
It was a while ago; I *think* that fixing `unix` got it to work though.

**Comment by atamon** [comment]
Alright, thanks for the lead! :)

**Comment by valderman** [comment]
Until this is fixed (and even after that - this build is finicky), it's highly recommended that you use the [pre-built binaries](http://haste-lang.org/downloads) for non-Linux platforms. Not only does this have a substantially higher likelihood of working, but you also get niceties such as manual pages and not having to wait forever for `haste-boot` to complete.

**Comment by saurabhnanda** [comment]
I have downloaded the pre-build Haste binary and am trying to compile https://github.com/yamafaktory/reflex-starter with it.

* `haste-cabal init` worked
* `haste-cabal install` is throwing a `/usr/bin/ar: permission denied`

**Comment by vilhelms** [comment]
Unfortunately, I don't think using the pre-build binary is a solution, because the haste-cabal that comes with it still doesn't seem to work:

```
$ haste-cabal install parsec
haste-cabal: /usr/bin/ar: permission denied
```

**Comment by valderman** [comment]
Fixed as of release 0.5.5.1.

**Comment by saurabhnanda** [comment]
So, time to try reflex-frp with haste now?
